js/plot/timechange.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2015, Tidepool Project
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ * 
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+var d3 = require('d3');
+var _ = require('lodash');
+
+var dt = require('../data/util/datetime');
+var format = require('../data/util/format');
+
+var timeChangeImage = require('../../img/timechange/timechange.svg');
+
+var log = require('bows')('DeviceEvent - TimeChange');
+
+/**
+ * Module for adding timechange markers to a chart pool
+ * 
+ * @param  {Object} pool the chart pool
+ * @param  {Object|null} opts configuration options
+ * @return {Object}      time change object
+ */
+module.exports = function(pool, opts) {
+  opts = opts || {};
+
+  /**
+   * Default configuration for this component
+   */
+  var defaults = {
+    highlightWidth: 4,
+    tooltipPadding: 20
+  };
+
+  _.defaults(opts, defaults);
+
+  var mainGroup = pool.group();
+
+  function timechange(selection) {
+    opts.xScale = pool.xScale().copy();
+
+    selection.each(function(currentData) {
+      var filteredData = _.filter(currentData, {subType: 'timeChange'});
+
+      var timechanges = d3.select(this)
+        .selectAll('g.d3-timechange-group')
+        .data(filteredData, function(d) {
+          return d.id;
+        });
+
+      var timechangeGroup = timechanges.enter()
+        .append('g')
+        .attr({
+          'class': 'd3-timechange-group',
+          id: function(d) {
+            return 'timechange_' + d.id;
+          }
+        });
+
+      timechange.addTimeChangeToPool(timechangeGroup);
+
+      timechanges.exit().remove();
+    });
+  }
+
+  timechange.addTimeChangeToPool = function(selection) {
+    opts.xScale = pool.xScale().copy();
+    selection.append('image')
+      .attr({
+        'xlink:href': timeChangeImage,
+        cursor: 'pointer',
```

Comment:
We did `pointer` for messages because they're clickable. Remove since these aren't?